client/v2/client.go

```diff
@@ -107,9 +107,9 @@ func (c *Client) Close() error {
 
 // Send attempts to JSON-encode and send all events without waiting for ACK.
 // Returns error if sending or serialization fails.
-func (c *Client) Send(data []interface{}) error {
+func (c *Client) Send(data []interface{}) (dropped int, err error) {
```

Comment:
With this signature it is difficult (if even possible) for the caller to know what events have been dropped, what may be useful to decide how to handle or report the errors.

What do you think about implementing this method in a similar fashion to `io.Writer`s? With an interface like `Send(data []interface{}) (n int, err error)`, being `n` the number of events sent, and `err` the error that interrupted the operation.
`Send` could serialize and send event by event, and if any event cannot be serialized or sent, `Send` is interrupted returning the number of events sent so far, and the reason of the failure for the last event.
I would find this way a better one to report errors to the caller, so they can decide how to handle the problem and how to continue, without needing additional options like this `failSerializationError`. This would be also more consistent with other well-known I/O APIs.
This would also allow us to avoid having to serialize the whole batch before sending it, reducing use of memory.